homeassistant/components/incomfort/config_flow.py

```diff
@@ -0,0 +1,94 @@
+"""Config flow support for Intergas InComfort integration."""
+
+from typing import Any
+
+from aiohttp import ClientResponseError
+from incomfortclient import IncomfortError, InvalidHeaterList
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import DOMAIN
+from .models import async_connect_gateway
+
+TITLE = "Intergas InComfort/Intouch Lan2RF gateway"
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): TextSelector(
+            TextSelectorConfig(type=TextSelectorType.TEXT)
+        ),
+        vol.Optional(CONF_USERNAME): TextSelector(
+            TextSelectorConfig(type=TextSelectorType.TEXT, autocomplete="admin")
+        ),
+        vol.Optional(CONF_PASSWORD): TextSelector(
+            TextSelectorConfig(type=TextSelectorType.PASSWORD)
+        ),
+    }
+)
+
+ERROR_STATUS_MAPPING: dict[int, tuple[str, str]] = {
+    401: (CONF_PASSWORD, "auth_error"),
+    404: ("base", "not_found"),
+}
+
+
+async def async_try_connect_gateway(
+    hass: HomeAssistant, config: dict[str, Any], errors: dict[str, str]
+) -> bool:
```

Comment:
Instead of modifying the errors dict (which is a sideeffect), should we instead just return the dict of errors?